exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -53,6 +54,60 @@ func TestMetricsConsumerNormal(t *testing.T) {
 	assert.Equal(t, 1, sender.numCloseCalls)
 }
 
+func TestMetricsConsumerNormalWithSourceTag(t *testing.T) {
+	sum := newMetric("sum", pdata.MetricDataTypeSum)
+	mockSumConsumer := &mockTypedMetricConsumer{typ: pdata.MetricDataTypeSum}
+	sender := &mockFlushCloser{}
+	tags := map[string]string{"source": "test_source", "test_key": "test_value"}
+	metrics := constructMetricsWithTags(tags, sum)
+	consumer := newMetricsConsumer(
+		[]typedMetricConsumer{mockSumConsumer}, sender, true)
+
+	assert.NoError(t, consumer.Consume(context.Background(), metrics))
+
+	assert.ElementsMatch(t, []string{"sum"}, mockSumConsumer.names)
+	assert.ElementsMatch(t, []string{"test_source"}, mockSumConsumer.sources)
+
+	assert.True(t, containsStr(mockSumConsumer.attrMaps, "test_key:test_value"))
+	assert.False(t, containsStr(mockSumConsumer.attrMaps, "source:test_source"))
+
+	assert.Equal(t, 1, mockSumConsumer.pushInternalMetricsCallCount)
+	assert.Equal(t, 1, sender.numFlushCalls)
+	assert.Equal(t, 0, sender.numCloseCalls)
+
+	consumer.Close()
+	assert.Equal(t, 1, sender.numCloseCalls)
+}
+
+func TestMetricsConsumerNormalWithHostnameTag(t *testing.T) {
+	sum := newMetric("sum", pdata.MetricDataTypeSum)
+	mockSumConsumer := &mockTypedMetricConsumer{typ: pdata.MetricDataTypeSum}
+	sender := &mockFlushCloser{}
+	tags := map[string]string{"host.name": "test_host.name", "hostname": "test_hostname"}
+	metrics := constructMetricsWithTags(tags, sum)
+	consumer := newMetricsConsumer(
+		[]typedMetricConsumer{mockSumConsumer}, sender, true)
+
+	assert.NoError(t, consumer.Consume(context.Background(), metrics))
+
+	assert.ElementsMatch(t, []string{"sum"}, mockSumConsumer.names)
+	assert.ElementsMatch(t, []string{"test_host.name"}, mockSumConsumer.sources)
+	assert.True(t, containsStr(mockSumConsumer.attrMaps, "hostname:test_hostname"))
+	assert.False(t, containsStr(mockSumConsumer.attrMaps, "host.name:test_host.name"))
```

Comment:
You can convert an AttributeMap to a map[string]string, but it takes some code:

```
func toGoMap(attrMap pdata.AttributeMap) map[string]string {
    result := make(map[string]string)
    attrMap.Range(func(key string, value AttributeValue) bool {
        result[key] = value.AsString()
        return true
    })
    return result
}
```

Then you can make the `attrMaps` attribute be a `[]map[string]string`. Then you can use the usual testify methods with it.

Reply:
resolved